components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,133 @@
+import { Dropdown } from "components";
+import { grey800, white } from "constants/colors";
+import { addOpacityToHsl } from "helpers";
+import { usePaginationContext } from "hooks";
+import PreviousPage from "public/assets/icons/left-chevron.svg";
+import NextPage from "public/assets/icons/right-chevron.svg";
+import styled, { CSSProperties } from "styled-components";
+import { DropdownItemT, PaginateForT } from "types";
+
+export interface Props {
+  paginateFor: PaginateForT;
+}
+
+export function Pagination({ paginateFor }: Props) {
+  const { pageStates, goToPage, nextPage, previousPage, setResultsPerPage } = usePaginationContext();
+
+  const pageState = pageStates[paginateFor];
+  const numberOfButtons = 5;
+  const numbersPastMax = pageState.number - numberOfButtons;
+  const buttonNumbers = Array.from({ length: numberOfButtons }, (_, i) => i + 1).map((number) => {
+    if (numbersPastMax > 0) {
+      return number + numbersPastMax;
+    }
+    return number;
+  });
+
+  const _goToPage = (page: number) => goToPage(paginateFor, page);
```

Comment:
convenience functions that use partial application of args to avoid having to specify `paginateFor` every time